### ISSUE TITLE
AWS ruleset fails on provider::kubernetes based functions

### ISSUE THREAD

**Issue by Skaronator** [terraform-linters/tflint-ruleset-aws#749]
## Issue
tflint throws an aws ruleset error on my kubernetes provider based functions:

```
12:49:19.458 INFO   Executing hook: tflint
Plugin `aws` is already installed
Plugin `terraform` is already installed
Failed to check ruleset; failed to check "aws_iam_policy_too_long_policy" rule: .terragrunt-cache/7pBeIgNM1J8RqJ9MsLSaIGFm3QQ/_ubfDZBV8nHg5hBv2BhoO2J0nVc/network/istio.tf:18,29-72: Call to unknown function; There are no functions in namespace "provider::kubernetes::".
12:49:19.574 ERROR  Error running hook tflint with message: Error while running tflint with args: [tflint --init --config ../../../.tflint.hcl --chdir ./.terragrunt-cache/7pBeIgNM1J8RqJ9MsLSaIGFm3QQ/_ubfDZBV8nHg5hBv2BhoO2J0nVc/network]
12:49:19.574 ERROR  Errors encountered running before_hooks. Not running 'terraform'.
12:49:19.574 ERROR  error occurred:

* Error while running tflint with args: [tflint --init --config ../../../.tflint.hcl --chdir ./.terragrunt-cache/7pBeIgNM1J8RqJ9MsLSaIGFm3QQ/_ubfDZBV8nHg5hBv2BhoO2J0nVc/network]
```

The code behind works just fine in Terraform (tofu) and tflint can correctly validate that code. I only got the error once I added an AWS specific resource. I guess that at that point to ruleset gets really activated.

Here is the code:

```hcl
data "http" "gateway_crds" {
  url = "https://github.com/kubernetes-sigs/gateway-api/releases/download/v1.2.0/standard-install.yaml"
  retry {
    attempts = 2
  }
}

resource "kubernetes_manifest" "gateway_crds" {
  for_each = { for value in provider::kubernetes::manifest_decode_multi(data.http.gateway_crds.body) : value.metadata.name => value }
  # we have to do this block in order to remove the "status" field
  manifest = {
    apiVersion = each.value.apiVersion
    kind       = each.value.kind
    # we have to do this block in order to remove the "creationTimestamp" field
    metadata = {
      name        = each.value.metadata.name
      annotations = each.value.metadata.annotations
    }
    spec = each.value.spec
  }
}
```

## versions

```bash
tflint --version 
TFLint version 0.54.0
+ ruleset.terraform (0.10.0-bundled)
```
with
```hcl
plugin "aws" {
  enabled = true
  source  = "github.com/terraform-linters/tflint-ruleset-aws"
  version = "0.34.0"
}

plugin "terraform" {
  enabled = true
  source  = "github.com/terraform-linters/tflint-ruleset-terraform"
  version = "0.10.0"
}
```

## addtional infos

Trying to ignore that error on that resource and/or the whole file didn't work either. I tried

```hcl
resource "kubernetes_manifest" "gateway_crds" {
  # tflint-ignore: aws_iam_policy_sid_invalid_characters
  for_each = { for value in provider::kubernetes::manifest_decode_multi(data.http.gateway_crds.body) : value.metadata.name => value }
```

and

```hcl
# tflint-ignore: aws_iam_policy_sid_invalid_characters
resource "kubernetes_manifest" "gateway_crds" {
  for_each = { for value in provider::kubernetes::manifest_decode_multi(data.http.gateway_crds.body) : value.metadata.name => value }
```

as well as 

```hcl
# tflint-ignore-file: aws_iam_policy_sid_invalid_characters
```
at the very top of the file.

ultimately setting this in the `.tflint.hcl` worked but is not good enough:

```hcl
rule "aws_iam_policy_sid_invalid_characters" {
  enabled = false
}

rule "aws_iam_policy_too_long_policy" {
  enabled = false
}

rule "aws_iam_policy_invalid_description" {
  enabled = false
}

rule "aws_iam_policy_invalid_name" {
  enabled = false
}

rule "aws_iam_policy_invalid_path" {
  enabled = false
}

rule "aws_iam_policy_invalid_policy" {
  enabled = false
}
```

**Comment by wata727** [comment]
Hmm, I couldn't reproduce the error in my environment. TFLint 0.54 supports provider-defined functions, so you should not get any errors. Can you provide a minimal code snippet that reproduces the issue?

The `tflint-ignore` annotation won't work for this error because it suppresses issues, not errors.

**Comment by Skaronator** [comment]
The error happend once I added an AWS resource (aws_iam_policy) in a different file but same tflint scope. On it's own the code works just fine in tflint.

I'll get a minimal code snippet that reproduces the issue tomorrow.

Thanks
